File: 961.go
func repeatedNTimes(nums []int) int {
    var idx int
    for idx + 2 < len(nums) {
        if nums[idx] == nums[idx+1] || nums[idx] == nums[idx+2] {
            return nums[idx]
        }
        if nums[idx+1] == nums[idx+2] {
            return nums[idx+1]
        }
        idx += 3
    }

    return nums[len(nums)-1]
}

File: 961.rs
impl Solution {
    pub fn repeated_n_times(nums: Vec<i32>) -> i32 {
        let n = nums.len() / 3;

        for i in 0..n {
            if nums[3*i] == nums[3*i+1] || nums[3*i] == nums[3*i+2] {
                return nums[3*i];
            }
            if nums[3*i+1] == nums[3*i+2] {
                return nums[3*i+1];
            }
        }

        nums[nums.len()-1]
    }
}
